File: src/main/kotlin/com/githubyss/mobile/common/ui/mvvm/binding_adapter/ImageViewBindingAdapter.java
// package com.githubyss.mobile.common.ui.mvvm.bindingadapter;
//
// import android.widget.ImageView;
//
// import com.githubyss.mobile.common.kit.glide.GlideUtils;
//
// import androidx.databinding.BindingAdapter;
//
//
// public class ImageViewBindingAdapter {
//
//     @BindingAdapter({"path"})
//     public static void loadImage(ImageView view, Object path) {
//         GlideUtils.INSTANCE.loadImage(view, view, path, null, null);
//     }
//
//     @BindingAdapter(value = {"path", "placeholder", "error"}, requireAll = false)
//     public static void loadImage(ImageView view, Object path, Object placeholder, Object error) {
//         GlideUtils.INSTANCE.loadImage(view, view, path, placeholder, error);
//     }
// }

File: src/main/kotlin/com/githubyss/mobile/common/ui/mvvm/binding_adapter/ImageViewBindingAdapter.kt
package com.githubyss.mobile.common.ui.mvvm.binding_adapter

import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.githubyss.mobile.common.kit.glide.GlideUtils


@BindingAdapter(value = ["bind:path", "bind:placeholder", "bind:error"], requireAll = true)
fun ImageView.loadImage(path: Any?, placeholder: Any?, error: Any?) {
    GlideUtils.loadImage(this, this, path, placeholder, error)
}
